src/read_preference.ts

```diff
@@ -164,36 +164,6 @@ export class ReadPreference {
     return readPreference as ReadPreference;
   }
 
```

Comment:
Note: `ReadPreference.fromOptions` should not default to `primary` if it cannot find a read pref in the options. In `resolveOptions` when `ReadPreference.fromOptions` is undefined, we default to the parent read pref. If `fromOptions` always defaults to primary, we would never use the parent read pref.